src/main/java/com/rabbitmq/http/client/Client.java

```diff
@@ -614,7 +665,11 @@ private String encodePathSegment(final String pathSegment) {
     return xs;
   }
 
-  private HttpComponentsClientHttpRequestFactory getRequestFactory(final URL url, final String username, final String password, final SSLConnectionSocketFactory sslConnectionSocketFactory, final SSLContext sslContext) throws MalformedURLException {
+  private HttpComponentsClientHttpRequestFactory getRequestFactory(final URL url,
+                                                                   final String username, final String password,
+                                                                   final SSLConnectionSocketFactory sslConnectionSocketFactory,
+                                                                   final SSLContext sslContext,
+                                                                   final HttpClientBuilderConfigurator configurator) throws MalformedURLException {
```

Comment:
I will explore this but we don't use `HttpEntity` and switching to it would be a lot of work for us to do to allow some post-configuration.